x/axelarnet/types/types.go

```diff
@@ -95,14 +97,21 @@ func SortTransfers(transfers []IBCTransfer) {
 
 // Validate checks the stateless validity of the cosmos chain
 func (m CosmosChain) Validate() error {
-	if m.Name != ModuleName && m.IBCPath == "" {
-		return fmt.Errorf("IBC path is empty")
+	err := utils.ValidateString(m.IBCPath)
+	if !strings.EqualFold(m.Name, exported.Axelarnet.Name) && err != nil {
+		return fmt.Errorf("invalid IBC path")
```

Comment:
Why `EqualFold` here?  Also, I don't think this makes sense. We only want to allow empty IBCPath if the chain's name is axelarnet, not that we want to allow the IBCPath to be an "invalid" string if the chain's name is axelarnet. For example, no error will be returned now if chain's name is axelarnet and IBCPath is "_".

Reply:
The previous logic was not allowing IBC path to be empty _if_ the chain name wasn't `axelarnet`. If it was, then there were no constraints on it. But I like the idea of enforcing it to be `""` for `axelarnet`.

Reply:
Also, same reason for `EqualFold`. We have been using that to compare chain names. Any thoughts @haiyizxx?

Reply:
Why was not it? 
```
if m.Name != ModuleName && m.IBCPath == "" {
		return fmt.Errorf("IBC path is empty")
```
It was saying that if `m.Name` is not `ModjuleName`, then `m.IBCPath` must be non-empty.

Reply:
If we do `EqualFold` in validation, we probably wanna make sure to lower case it when saving it to the kvstore. Then we should just allow it to be strictly equal to "axelarnet".

Reply:
There's a not operator `!` before `EqualFold`. So the logic was similar, if `m.Name` is not equal to Axelarnet, then IBC path must be valid. But I have updated it to also require IBCPath to be empty if it is `axelarnet`.

Reply:
We do lower case it everywhere from what I've seen. But in most places we allow the request chain name to be of different case, so keeping it consistent.

Reply:
it is related to the default genesis change
```
Chains: []CosmosChain{{
	Name:       ModuleName,                        <-before 'axelarnet'
	Name:       exported.Axelarnet.Name,   <- changed in this pr 'Axelarnet'
	AddrPrefix: "axelar",
}}
```